### PR TITLE
[ROCm] Proper fix for XLA unit test seg fault issue.

### PR DESCRIPTION
**Pull request by rsanthanam-amd**
This was root caused to be a divergence between the CUDA and HIP
conv_canonicalizer implementation.

### CODE REVIEW

**Comment by rsanthanam-amd** [comment]
/cc @cheshire @jurahul 

I found the root cause of the seg fault with the original change in gpu_compiler.cc.

This PR fixes the seg fault and restores that original change.

File: tensorflow/compiler/xla/service/gpu/gpu_compiler.cc [hunk 1]
```diff
@@ -395,7 +395,7 @@ Status GpuCompiler::OptimizeHloModule(
     AlgebraicSimplifierOptions options;
     options.set_replace_transpose_with_bitcast(false);
     options.set_enable_conv_operand_swap(false);
-    collectives_pipeline.AddPass<HloPassFix<AlgebraicSimplifier>>(options);
+    collectives_pipeline.AddPass<AlgebraicSimplifier>(options);
```
Comment (cheshire) [inline tensorflow/compiler/xla/service/gpu/gpu_compiler.cc:398]: Do we need this change?

**Comment by rsanthanam-amd** [comment]
that's the way it was originally written but the HloPassFix was added as a temporary fix to enable it to work with HIP.

I worked with @jurahul on this.

But this PR restores it to the original way it was implemented by fixing HIP to work with that.

File: tensorflow/compiler/xla/service/gpu/gpu_compiler.cc (hunk 1 above)
Comment (jurahul) [reply tensorflow/compiler/xla/service/gpu/gpu_compiler.cc:398]: The PassFix<> was added to fix AMD HIP issues. If the change in AMDGPUCompiler::OptimizeHloConvolutionCanonicalization eliminates the need for that, I am ok removing the PassFix.

**Review by jurahul — APPROVED** [review]
LGTM

**Review by gbaned — APPROVED** [review]

**Pull request merged**